webui/react/src/components/DeterminedAuth.tsx

```diff
@@ -1,10 +1,10 @@
 import { Button, Form, Input } from 'antd';
-import React, { useState } from 'react';
+import React, { useCallback, useState } from 'react';
 
 import Icon from 'components/Icon';
-import Auth, { updateAuth } from 'contexts/Auth';
+import Auth from 'contexts/Auth';
 import handleError, { ErrorType } from 'ErrorHandler';
-import { isLoginFailure, login } from 'services/api';
+import { getCurrentUser, isLoginFailure, login } from 'services/api';
```

Comment:
I believe so, but let me run through the two use cases to make sure. The one in `SignIn` is used for detecting whether the user is already authenticated and then redirect the user to the app if it is the case. The one in `DeterminedAuth` is used for the user triggered sign in via username and password. The `SignIn` is needed at this level I presume for when OAuth and SAML are added on, they will also need this to check their verification. It might be possible to have `DeterminedAuth` to retrigger the auth check in `SignIn` upon a successful login, but I don't have a straight forward implementation of that at this moment.

Reply:
> A lot of good ideas here + a nice oneliner replacement for AuthToken component.
> Below is what I ran into during testing, some purely stylistic preferences :)
> 
> * an issue probably around logout detection. a loop. happens on master as well. repro: open the dashboard in two tabs and log out from one. the other one would go in an infinite loop.

This should be addressed now with the recent commits.

> * the login page shows up for a short period during page reloads. I guess it'd show up between react route changes if we had them as well.

This is a good find, and it has to do with auth data and spinner data being off sync. I addressed this in the recent PR as well.

> * react state update hook error ( same as the one from PR description, am I using an old build? 🤔 ) this happens on normal page-load of the dashboard in a signed-in state

It wasn't completely cleaned up previously, found more cases that triggered it. This is mostly cleaned up, all but one remaining that I found. The last one I left for a separate PR, where the auto sign out causes a change in state to an unmounted component. So if you have two sessions, both in logged in state, when signing out in the first one, the second one will do so as well when it attempts to poll for stats the next time. The 2nd one will trigger the error.

> * style: I prefer the bigger scale? (size) of the login page, it's the only thing on the page.

I'm hoping to unify form elements across the entire webui and as part of that is to use a consistent treatments, colors and sizing.

> * the disabled sign-in button has on hover effect. is that intentional?

Great find, also was missing styling for active, focus and other states.

> * nit: ideally sign-in would be disabled until username or password has changed after a bad credentials error.

Agreed. Briefly looked at this and will leave this for a different PR. It gets into modifying antd behavior so it might become a larger body of work than anticipated.

Reply:
> > * style: I prefer the bigger scale? (size) of the login page, it's the only thing on the page.
> 
> I'm hoping to unify form elements across the entire webui and as part of that is to use a consistent treatments, colors and sizing.

I was initially tending towards a larger size but I do prefer consistent sizing overall. I'd like to not surprise the user with the (immediate) size change for all other elements after they login.

Reply:
> > The one in DeterminedAuth is used for the user triggered sign in via username and password.
> 
> to get the user after a successful login but not to check the current login status correct 🤔 ?

right, it would be nice if the login actually returned the auth user to avoid having to make an immediate 2nd api call to get the user.